### ISSUE TITLE
Memory leak / GC?

### ISSUE THREAD

**Issue by a-fent** [grosser/parallel#306]
Is there anything particular I need to do to make sure finished jobs are cleaned up and memory from them reclaimed? I am trying to process ~10,000 files in parallel and towards the end of the job the ruby process is using 55+GB of memory. It stays constant at ~0.5 GB when processing sequentially.

My parallel code looks like:

```
Parallel.map(SRC_FILES, in_threads: 4) do |src_file|
  process_file(src_file)
end
```

And the function simply accepts a file name and does something like

```
def process_file(src_file)
  # read gzip
  # load json
  # reshape json
  # post json to remote API
end
```
The remote API is the blocking operation I want to parallelise (it's an Elasticsearch bulk index). All the variables holding the large JSON structures are method-local so should fall out of scope and be eligible for garbage collection when the method returns.
